tests/unit/amo/components/TestSearchContextCard.js

```diff
@@ -111,162 +134,393 @@ describe(__filename, () => {
+    const { store } = dispatchSearchResults({
+      filters: { category: 'causes' },
+    });
+
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = render({ store });
+
+    sinon.assert.calledWith(
+      dispatchSpy,
+      fetchCategories({
+        errorHandlerId: root.instance().props.errorHandler.id,
+      }),
     );
   });
 
-  it('should render plural form when multiple results are found with addonType ADDON_TYPE_THEME', () => {
+  it('should not fetch categories if there is no category filter', () => {
     const { store } = dispatchSearchResults({
+      filters: { query: 'test' },
+    });
+
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    render({ store });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it("should not fetch categories if there is a category filter but it's already loading", () => {
+    const { store } = dispatchClientMetadata();
+
+    _searchStart({
+      store,
       filters: {
-        addonType: ADDON_TYPE_THEME,
+        category: 'causes',
+      },
+    });
+
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    render({ store });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('should render results with categoryName and query for addonType ADDON_TYPE_THEMES_FILTER when loading is false', () => {
+    const categoryName = 'Causes';
+
+    const { store } = dispatchSearchResults({
+      filters: {
+        addonType: ADDON_TYPE_THEMES_FILTER,
+        category: 'causes',
         query: 'test',
       },
     });
 
+    _fetchCategories({ store });
+    _loadCategories({ store });
+
+    const root = render({ store });
+
+    expect(root.find('.SearchContextCard-header')).toIncludeText(
+      `2 themes found for "test" in ${categoryName}`,
+    );
+  });
+
+  it('should render results with categoryName and no query for addonType ADDON_TYPE_THEMES_FILTER when there is no query and loading is false', () => {
+    const categoryName = 'Causes';
+
+    const { store } = dispatchSearchResults({
+      filters: {
+        addonType: ADDON_TYPE_THEMES_FILTER,
+        category: 'causes',
+      },
+    });
+
+    _fetchCategories({ store });
+    _loadCategories({ store });
+
     const root = render({ store });
 
     expect(root.find('.SearchContextCard-header')).toIncludeText(
-      '2 themes found for "test"',
+      `2 themes found in ${categoryName}`,
+    );
+  });
+
+  it('should render results without categoryName or query when neither are present for addonType ADDON_TYPE_THEMES_FILTER', () => {
+    const { store } = dispatchSearchResults({
+      filters: {
+        addonType: ADDON_TYPE_THEMES_FILTER,
+      },
+    });
+
+    const root = render({ store });
+
+    expect(root.find('.SearchContextCard-header')).toIncludeText(
+      '2 themes found',
     );
   });
 
   it('should render singular form when only one result is found with addonType ADDON_TYPE_DICT', () => {
```

Comment:
Do we still need tests specifically for `ADDON_TYPE_DICT` when that is no longer being specifically dealt with in the code? Should we instead just have a set of tests for when the type is neither extension nor theme?

Reply:
good point. I think I was just showing as a non extension/theme examples but I think that might be covered. I'll double check and remove or make this more generic.